ctk/common/src/main/kotlin/org/codice/compliance/verification/core/CoreVerifier.kt

```diff
@@ -130,13 +134,37 @@ class CoreVerifier(val node: Node) {
                     responseDom.recursiveChildren("EncryptedAttribute") +
                     responseDom.recursiveChildren("EncryptedID")
         }
+
+        private fun validateEntityIdentifiers(responseDom: Node) {
+            responseDom.recursiveChildren().filter { it.attributeText("Format") == ENTITY }
+                    .forEach { checkEntityIdentifier(it) }
+        }
+
+        private fun checkEntityIdentifier(node: Node) {
+            if (node.attributeNode("NameQualifier") != null ||
+                    node.attributeNode("SPNameQualifier") != null ||
+                    node.attributeNode("SPProvidedID") != null) {
+                throw SAMLComplianceException.create(SAMLCore_8_3_6_a,
+                        message = "No Subject element found.",
+                        node = node)
+            }
+            val nodeValue = node.nodeValue
```

Comment:
✏️ Alternatively:
```
node.nodeValue?.let {
    if (it.length > ENTITY_ID_MAX_LEN)
        throw SAMLComplianceException.create(SAMLCore_8_3_6_b,
            message = "Length of URI [$it] is [${it.value}]",
            node = node)
}
```

Reply:
Should have seen that when I pulled my conditions apart and extracted the `nodeValue` but I completely missed it. Nice.